spec/models/post_spec.rb

```diff
@@ -35,5 +35,16 @@
         expect(post.up_votes).to eq(1)
       end
     end
+
+    describe "#save_with_initial_vote" do
+      it "saves a post only if the initial vote saves as well" do
+        post = described_class.new(title: "Check One Two",
+                        body:  "Sibilance. Sibilance.",
+                        user:  authenticated_user,
+                        created_at: Time.new)
```

Comment:
@npauzenga this is definitely good :smile:! You are right, you can't use `associated_post` because it saves the post before you can test `save_with_initial_vote`. You could also create another method like you said:

```ruby
def associated_post_unsaved(options = {})
  post_options = {
    title: "Post title",
    body:  "Post bodies must be pretty long.",
    topic: Topic.create(name: "Topic name"),
    user:  authenticated_user
  }.merge(options)

  Post.new(post_options)
end

def associated_post(options = {})
  post = associated_post_unsaved(options)
  post.save
  post
end

# or

# tap is a nice little method that yields and returns
# the object that calls it.
def associated_post(options = {})
  associated_post_unsaved(options).tap { |post| post.save }
end
```

I'm definitely a fan of reducing the amount of code in tests to only the actual tests (like removing the creation/configuration of test objects and such). 

You can also use the [Fabrication gem](http://www.fabricationgem.org/), which I'm a big fan of :smiley:. It helps you define your test factories, so you don't need to define all these methods yourself.

Also, if you want to see the documentation on `tap`: http://ruby-doc.org/core-2.2.3/Object.html#method-i-tap